### ISSUE TITLE
confirmation about usage of inert_defer

### ISSUE THREAD

**Issue by cyrus86** [realm/SwiftLint#2611]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

inert_defer is being triggered if the defer is at the end of the parent scope. However there's at least one case where this actually yields a different result than running the statement without defer.

If you want to fire a `didSet` of a class property from initializer, putting it into `defer` will do that for you. Without the `defer` at the end of the initializer scope, `didSet` will not fire. 

### Environment

* SwiftLint version: 0.30.1
* Installation method used: Homebrew
* Paste your configuration file:

```yml
disabled_rules:
- line_length
- todo
- variable_name

- force_try
- valid_docs
- force_cast
- type_body_length
- function_body_length
- file_length
- void_return
- function_parameter_count
- valid_ibinspectable
- cyclomatic_complexity

excluded:
 - Pods

line_length: 120;
```

* Which Xcode version are you using: 10.1

```swift
// This triggers a violation:
init() {
    // other init stuff
    defer {
        self.state = 1
    }
}

var state: Int = 1 {
    didSet {
        // do something
    }
}

```

**Comment by jpsim** [comment]
I recall someone else bringing this up. I'd consider this a false positive in the rule.

Swift 5 presumably adds this warning to the compiler itself. I'd be interested in knowing if it also warns in this case. I suspect it does.

**Comment by marcelofabri** [comment]
I don’t think we should treat it as a bug. IIRC the current behavior on inits is not something that was planned on the compiler conscientiously.

I’d prefer this to be a configuration on the rule.

**Comment by marcelofabri** [comment]
Finally found where I had seen this being categorized as a bug: https://twitter.com/jckarter/status/926459181661536256

**Comment by jpsim** [comment]
Ok, that's pretty definitive. Let's close this then. Thanks for doing the digging @marcelofabri!

**Comment by cyrus86** [comment]
thanks! That cleared things up for me. Gonna use a helper function then.
thanks @marcelofabri !
